source/common/common/cleanup.h

```diff
@@ -64,8 +64,11 @@ template <class Key, class Value> class RaiiMapOfListElement {
 
   template <typename ConvertibleToKey>
   RaiiMapOfListElement(MapOfList& map, const ConvertibleToKey& key, Value value)
-      : map_(map), list_(map_.try_emplace(key).first->second), key_(key), cancelled_(false) {
-    it_ = list_.emplace(list_.begin(), value);
+      : map_(map), key_(key), cancelled_(false) {
+    // The list reference itself cannot be saved because it is not stable in the event of a
```

Comment:
note: an alternative fix is to use absl::node_hash_map. I went ahead with this fix because it doesn't require usage site changes, and abseil generally recommends using flat_hash_map, over node_hash_map unless it's necessary.